tests/rptest/tests/archival_test.py

```diff
@@ -203,14 +203,26 @@ def test_write(self):
         self.kafka_tools.produce(self.topic, 10000, 1024)
         validate(self._quick_verify, self.logger, 90)
 
-    @ignore  # https://github.com/vectorizedio/redpanda/issues/1977
     @cluster(num_nodes=3)
     def test_isolate(self):
         """Verify that our isolate/rejoin facilities actually work"""
         with firewall_blocked(self.redpanda.nodes, self._get_s3_endpoint_ip()):
             self.kafka_tools.produce(self.topic, 10000, 1024)
             time.sleep(10)  # can't busy wait here
-            self._check_bucket_is_emtpy()
+
+            # Topic manifest can be present in the bucket because topic is created before
+            # firewall is blocked. No segments or partition manifest should be present.
+            topic_manifest_id = "d0000000/meta/kafka/panda-topic/topic_manifest.json"
+            objects = self.s3_client.list_objects(self.s3_bucket_name)
+            keys = [x.Key for x in objects]
+            self.logger.debug(f"keys: {keys}")
+
+            assert len(keys) < 2, \
+                f"Bucket should be empty or contain only {topic_manifest_id}, but contains {keys}"
```

Comment:
based on the assertion comment, it looks like the test on `keys` can be stronger by inspecting it to see if it contains `topic_manifest_id`. as currently written it seems like the assertion could pass if list_objects returned a single key and that key happened to not be the expected `topic_manifest_id`.